File: backend/app/schemas/github.py
from datetime import datetime
from typing import Annotated
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.resume import JobPollingResponse


class GitHubRepositoryConnectRequest(BaseModel):
    model_config = ConfigDict(extra="forbid", str_strip_whitespace=True)

    repository_url: Annotated[str, Field(min_length=1, max_length=2048)]


class GitHubRepositoryResponse(BaseModel):
    id: UUID
    repository_url: str
    created_at: datetime
    job: JobPollingResponse | None


class GitHubRepositorySnapshotSummary(BaseModel):
    description: str | None
    is_archived: bool | None
    languages: list[str]
    file_count: int
    manifest_count: int


class GitHubRepositorySkillResponse(BaseModel):
    name: str
    category: str
    extraction_confidence: float


class GitHubRepositoryDetailResponse(GitHubRepositoryResponse):
    snapshot: GitHubRepositorySnapshotSummary | None
    skills: list[GitHubRepositorySkillResponse]

File: frontend/lib/api/types/github.ts
import type { JobPollingResponse } from "@/lib/api/types/jobs";

export type GitHubRepositoryResponse = {
  id: string;
  repository_url: string;
  created_at: string;
  job: JobPollingResponse | null;
};

export type GitHubRepositorySnapshotSummary = {
  description: string | null;
  is_archived: boolean | null;
  languages: string[];
  file_count: number;
  manifest_count: number;
};

export type GitHubRepositorySkill = {
  name: string;
  category: string;
  extraction_confidence: number;
};

export type GitHubRepositoryDetailResponse = GitHubRepositoryResponse & {
  snapshot: GitHubRepositorySnapshotSummary | null;
  skills: GitHubRepositorySkill[];
};

export type GitHubRepositoryConnectRequest = {
  repository_url: string;
};
